### PR TITLE
2024 Q4 Hyperledger Fabric report

### PR DESCRIPTION
**Pull request by denyeart**
2024 Q4 Hyperledger Fabric report.

### CODE REVIEW

File: tac/project-updates/2024/2024-Q4-Hyperledger-Fabric.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
+---
+layout: default
+title: 2024 Q3 Hyperledger Fabric
```
Comment (tkuhrt) [inline tac/project-updates/2024/2024-Q4-Hyperledger-Fabric.md:3]:
```suggestion
title: 2024 Q4 Hyperledger Fabric
```

**Review by tkuhrt — APPROVED** [review]
Congratulations on the v3.0 and BFT consensus!

File: tac/project-updates/2024/2024-Q4-Hyperledger-Fabric.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+The IBM maintainers have been less active recently. If there are users or developers interested in becoming more active contributors, and eventually maintainers, the project could use your help!
+
+# Contributor Diversity
+
+- 51% of code activities were performed by top 3 contributors. Remaining 49% performed by 101 other contributors.
+- IBM contributors accounted for 83% of activity.
+- Of top 10 contributors, 6 of 10 from IBM.
+
+# Additional Information
+
```
Comment (yacovm) [inline tac/project-updates/2024/2024-Q4-Hyperledger-Fabric.md:60]: Roadmap? Now that we have released V3.0 - what's next?

**Review by yacovm — APPROVED** [review]

**Comment by yacovm** [comment]
> Congratulations on the v3.0 and BFT consensus!

Thanks ;-)

File: tac/project-updates/2024/2024-Q4-Hyperledger-Fabric.md (hunk 2 above)
Comment (denyeart) [reply tac/project-updates/2024/2024-Q4-Hyperledger-Fabric.md:60]: I think the project is at a state of maturity where the current maintainer's 'must-dos' and low-hanging fruit relative to our dev bandwith has largely been done. On the wiki roadmap page we have potential items like:

- Binary chaincode builder - replace default docker builder
- Optimize block storage, e.g. de-duplication of certs
- Replace Goleveldb with faster database
- Validation/Commit path performance improvements
- Orderer snapshots - similar to peer snapshots introduced in v2.3
- Block archive and prune - peer and orderer
- Quantum safe crypto

Each of those will require significant design and dev investment that is not currently planned, and they should be balanced with other community requests. I think the reality is that what is worked next will be dictated by who can contribute the developer sweat equity, taking into consideration their priorities and alignment with maintainers ideas, as driven through the RFC process.

File: tac/project-updates/2024/2024-Q4-Hyperledger-Fabric.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+New features get proposed, approved, and implemented based on a community RFC process.
+Mailing list activity is down a bit compared to prior years.
+PRs and mailing list questions are generally turned around quickly.
+There is significant Discord activity.
+
+# Questions/Issues for the TAC
+
+We would like to shift the project release images from DockerHub to Github Container Registry (ghcr). Note that the project release images are intended for trial use rather than production use, since they are not maintained regularly for security vulnerabilities, etc (production users should work with a vendor for commercially supported images). But still, there are a few concerns we'd like to raise for feedback.
+- Existing deployments and users may have scripts that pull images from DockerHub, for example as part of CI jobs. Would it be acceptable to break these scripts with the move?
+- We could leave all old images on DockerHub, and publish new images to Github Container Registry. Would this be confusing to users?
```
Comment (VRamakrishna) [inline tac/project-updates/2024/2024-Q4-Hyperledger-Fabric.md:21]: IMO this is perfectly fine as long as you have a note on the project's README page (and/or in the official docs) stating that you are doing this so legacy scripts don't break, while warning users that the Dockerhub images are out-of-date and the ghcr dependencies ought to be used instead.

**Review by VRamakrishna — APPROVED** [review]

**Review by yacovm — APPROVED** [review]

**Review by petermetz — APPROVED** [review]

**Review by mbrandenburger — APPROVED** [review]

**Review by jimthematrix — APPROVED** [review]
Congrats on the significant release v3.0 and making a BFT consensus available!

**Pull request merged**